core/src/engine/task/work_stealing_queue/consumer.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#include <moodycamel/blockingconcurrentqueue.h>
+#include <moodycamel/lightweightsemaphore.h>
+#include <array>
+#include <boost/smart_ptr/intrusive_ptr.hpp>
+#include <cstddef>
+#include <random>
+#include "engine/task/task_context.hpp"
+#include "local_queue.hpp"
+
+#include <engine/task/task_processor_config.hpp>
+
+constexpr std::size_t kLocalQueueMaxSize = 63;
+constexpr std::size_t kConsumerStealBufferSize = 32;
+constexpr std::size_t kStealAttempts = 4;
+
+USERVER_NAMESPACE_BEGIN
+namespace engine {
+
+class WorkStealingTaskQueue;
+class ConsumersManager;
+
+class Consumer final {
+ public:
+  Consumer();
+
+  Consumer(WorkStealingTaskQueue* owner, ConsumersManager* consumers_manager,
+           std::size_t inner_index);
+
+  Consumer(const Consumer& rhs);
+
+  bool Push(impl::TaskContext* ctx);
+
+  impl::TaskContext* Pop();
+
+  std::size_t LocalQueueSize() const;
```

Comment:
Scatter `noexcept` on all methods except for method that allocate or methods, which *might* reasonably start allocating in the future